src/str.c

```diff
@@ -51,6 +52,23 @@ coap_str_const_t *coap_make_str_const(const char *string)
   return &var[ofs];
 }
 
+coap_binary_t *coap_new_binary(size_t size) {
+  return (coap_binary_t *)coap_new_string(size);
+}
+
+coap_binary_t *coap_resize_binary(coap_binary_t *s, size_t size) {
+  coap_binary_t *new = realloc(s, sizeof(coap_binary_t) + size);
```

Comment:
We cannot just call `realloc()` here because the string object has not necessarily been allocated by `malloc`. At least, there should be a wrapper similar to `coap_malloc_type()` to avoid calling `realloc()` on invalid pointers. See also `mem.c`.

Reply:
Fair comment.  I did think about creating an new coap_binary_t and copying old to new and then deleting old.  I will get this properly reworked.

Reply:
Code updated and pushed